services/apps/src/apps/service/workspace.clj

```diff
@@ -1,9 +1,11 @@
 (ns apps.service.workspace
   (:use [korma.db :only [transaction]])
-  (:require [apps.persistence.workspace :as wp]))
+  (:require [apps.clients.iplant-groups :as iplant-groups]
+            [apps.persistence.workspace :as wp]))
 
 (defn get-workspace
-  [{:keys [username]}]
+  [{short-username :shortUsername :keys [username]}]
+  (iplant-groups/add-de-user short-username)
```

Comment:
Yeah, that's a good point. I put it here because it seemed somewhat related to creating a workspace. (No apps will be listed if the user either doesn't have a workspace or isn't in the de-users group.) Terrain doesn't currently have an iplant-groups client, but one could be added. I could also create a new endpoint in apps that ensures that the user is in the appropriate group. I think I actually prefer the latter option because the apps service has to know the appropriate group name anyway.